src/relay/relay.py

```diff
@@ -733,7 +735,10 @@ def _start_listen_network(self, address):
     def _start_push_service(self):
         logger.info("Start pushnotification service")
         path = self.config["push_notification"]["firebase_credentials_path"]
-        self._firebase_raw_push_service = FirebaseRawPushService(path)
+        cred = credentials.Certificate(path)
+        self._firebase_raw_push_service = FirebaseRawPushService(
+            firebase_admin.initialize_app(cred)
+        )
```

Comment:
Why do you think it is better to initialize the firebase app in `relay` than in `FirebaseRawPushService`?

Reply:
Yes, I was unsure about this, but couldn't find a better way than dependency injection. 

The problem is that in order to test the firebaseRawPushService I had to provide a path to the firebase config file.  In unit tests one needs to mock the firebase credentials. So my first implementation was - create the credentials object and pass it to the rawpushservice. Tehn however to further test the sending one needs to set a dummy project for the firebase_app and since the app is inside of the firebase push service I couldn't easily do that. So I decided to create the firebase app outside of the raw push service and inject it. This way I can test the implementation with unit tests. 

Better ideas?

Reply:
I created a helper function that can initialize the firebase app from the provided path. This way we are moving the credentials dependency outside of the relay.py.